1 Niagara 4 Overview Exercise - BFile/myFile-rt/src/com/tridiumuniversity/myFile/BFile.java

```diff
@@ -0,0 +1,220 @@
+
+  /**
+   * Get the {@code filename} property.
+   * @see #filename
+   */
+  public String getFilename() { return getString(filename); }
+
+  /**
+   * Set the {@code filename} property.
+   * @see #filename
+   */
+  public void setFilename(String v) { setString(filename, v, null); }
+
+  //endregion Property "filename"
+
+  //region Property "path"
+
+  /**
+   * Slot for the {@code path} property.
+   * @see #getPath
+   * @see #setPath
+   */
+  public static final Property path = newProperty(Flags.READONLY | Flags.SUMMARY, "", null);
+
+  /**
+   * Get the {@code path} property.
+   * @see #path
+   */
+  public String getPath() { return getString(path); }
+
+  /**
+   * Set the {@code path} property.
+   * @see #path
+   */
+  public void setPath(String v) { setString(path, v, null); }
+
+  //endregion Property "path"
+
+  //region Property "size"
+
+  /**
+   * Slot for the {@code size} property.
+   * @see #getSize
+   * @see #setSize
+   */
+  public static final Property size = newProperty(Flags.READONLY | Flags.SUMMARY, 0, null);
+
+  /**
+   * Get the {@code size} property.
+   * @see #size
+   */
+  public int getSize() { return getInt(size); }
+
+  /**
+   * Set the {@code size} property.
+   * @see #size
+   */
+  public void setSize(int v) { setInt(size, v, null); }
+
+  //endregion Property "size"
+
+  //region Property "createdDate"
+
+  /**
+   * Slot for the {@code createdDate} property.
+   * @see #getCreatedDate
+   * @see #setCreatedDate
+   */
+  public static final Property createdDate = newProperty(Flags.READONLY | Flags.SUMMARY, BAbsTime.DEFAULT, null);
+
+  /**
+   * Get the {@code createdDate} property.
+   * @see #createdDate
+   */
+  public BAbsTime getCreatedDate() { return (BAbsTime)get(createdDate); }
+
+  /**
+   * Set the {@code createdDate} property.
+   * @see #createdDate
+   */
+  public void setCreatedDate(BAbsTime v) { set(createdDate, v, null); }
+
+
+  //endregion Property "createdDate"
+
+  //region Property "modifiedDate"
+
+  /**
+   * Slot for the {@code modifiedDate} property.
+   * @see #getModifiedDate
+   * @see #setModifiedDate
+   */
+  public static final Property modifiedDate = newProperty(Flags.READONLY | Flags.SUMMARY, BAbsTime.DEFAULT, null);
+
+  /**
+   * Get the {@code modifiedDate} property.
+   * @see #modifiedDate
+   */
+  public BAbsTime getModifiedDate() { return (BAbsTime)get(modifiedDate); }
+
+  /**
+   * Set the {@code modifiedDate} property.
+   * @see #modifiedDate
+   */
+  public void setModifiedDate(BAbsTime v) { set(modifiedDate, v, null); }
+
+  //endregion Property "modifiedDate"
+
+  //region Action "touch"
+
+  /**
+   * Slot for the {@code touch} action.
+   * @see #touch()
+   */
+  public static final Action touch = newAction(Flags.SUMMARY, null);
+
+  /**
+   * Invoke the {@code touch} action.
+   * @see #touch
+   */
+  public void touch() { invoke(touch, null, null); }
+
+
+    public void doTouch()
```

Comment:
Your `doTouch` and `doPrint` methods look good, but remember that you should always put them outside of the code block that slotomatic generated for you. If another developer (or you in the future) were to change any of the annotations at the top of this class and run `gradlew slotomatic`, then slotomatic would overwrite this entire code block, deleting your `doTouch` and `doPrint` methods.